### PR TITLE
feat: Buffer readDiscard

### PR DESCRIPTION
**Pull request by turuslan**
`Buffer` class is used by parsers.
Sometimes parser needs to advance read position without copying bytes.
For example, zero-copy parser peeks available `bytes()`, then consumes some of them and advances read position.

`Buffer` class hides read position as `#off` private field.
The only function that advances read position is `read(p)`/`readSync(p)`, which accept buffer as argument.

```typescript
// currently available workaround
function readDiscard(buf: Buffer, n: number) {
  buf.readSync({ byteLength: n, set() {} });
}
```

Go provides [https://pkg.go.dev/bytes#Buffer.Next](Next(n)) function, which advances read position without copying bytes.
This PR proposes `readDiscard(n)` (please suggest better name), which doesn't return value.

There is `truncate(n)` function, which does opposite, it keeps "first N bytes", not "after N bytes"

### CODE REVIEW

File: io/buffer.ts [hunk 1]
```diff
@@ -183,6 +183,30 @@ export class Buffer implements Writer, WriterSync, Reader, ReaderSync {
...
+   * const buf = new Buffer();
+   * await buf.write(new TextEncoder().encode("Hello, world!"));
+   * buf.readDiscard(3);
+   * assertEquals(buf.length, 10);
+   * ```
+   *
+   * @param n The number of bytes to discard.
+   */
+  readDiscard(n: number) {
+    if (n < 0 || n > this.length) {
```
Comment (bartlomieju) [inline io/buffer.ts]: This guard misses non-integers. `readDiscard(NaN)` passes both comparisons — `NaN < 0` and `NaN > this.length` are both `false` — so `#off` becomes `NaN` permanently, which makes `length` `NaN` and silently corrupts the buffer with no error. Same for fractional `n`.

```suggestion
    if (!Number.isInteger(n) || n < 0 || n > this.length) {
```

File: io/buffer.ts [hunk 2]
```diff
@@ -183,6 +183,30 @@ export class Buffer implements Writer, WriterSync, Reader, ReaderSync {
...
+      throw new Error("Buffer read discard out of range");
```
Comment (bartlomieju) [inline io/buffer.ts]: Two things here, per the error-message style guide in [CONTRIBUTING.md](https://github.com/denoland/std/blob/main/.github/CONTRIBUTING.md#error-messages):

1. The message is passive and state-based rather than action-based, and it omits the offending value — the reader can't tell what `n` was or what range was expected.
2. `RangeError` is the right type for an out-of-range argument. (`truncate()` above uses bare `Error`, but that's pre-existing and new code shouldn't copy it.)

```suggestion
      throw new RangeError(
        `Cannot discard bytes as "n" must be an integer >= 0 and <= ${this.length}: received ${n}`,
      );
```

File: io/buffer.ts [hunk 3]
```diff
@@ -183,6 +183,30 @@ export class Buffer implements Writer, WriterSync, Reader, ReaderSync {
...
+   * import { Buffer } from "@std/io/buffer";
+   * import { assertEquals } from "@std/assert/equals";
+   *
+   * const buf = new Buffer();
+   * await buf.write(new TextEncoder().encode("Hello, world!"));
+   * buf.readDiscard(3);
+   * assertEquals(buf.length, 10);
+   * ```
+   *
+   * @param n The number of bytes to discard.
```
Comment (bartlomieju) [inline io/buffer.ts:201]: The prose above documents that this throws, so it should carry a `@throws` tag to match the rest of the module:

```suggestion
   * @param n The number of bytes to discard.
   * @throws {RangeError} If `n` is not an integer, is negative, or exceeds the buffer length.
```

File: io/buffer_test.ts [hunk 4]
```diff
@@ -400,3 +400,20 @@ Deno.test("Buffer.bytes() with copy set to false after Buffer.grow()", () => {
...
 });
+
+Deno.test("Buffer.readDiscard()", () => {
+  const bytes = Uint8Array.of(10, 11, 12, 13);
+  const reader = new Buffer();
+
+  writeAllSync(reader, bytes);
+  assertEquals(reader.bytes(), bytes.subarray(0, 4));
+
+  assertThrows(() => reader.readDiscard(5));
```
Comment (bartlomieju) [inline io/buffer_test.ts]: This asserts that *something* throws, but pins neither the error class nor the message — so it would still pass if the failure mode changed entirely. Repo tests normally pin both, e.g. `assertThrows(() => reader.readDiscard(5), RangeError, "Cannot discard bytes")`.

Could you also cover the edge cases? Negative `n`, `n === 0`, discarding from an empty buffer, and a non-integer such as `NaN` (see my note on the guard — that one currently passes silently).

**Review by bartlomieju — COMMENTED** [review]
Thanks for the PR, and for the clear motivation — the `buf.readSync({ byteLength: n, set() {} })` workaround is genuinely awkward, and a `bytes.Buffer.Next(n)`-style consume is a reasonable thing to want.

The blocker isn't the code, it's the venue. `@std/io` is in wind-down: `io/mod.ts` states that the `Reader`/`Writer` interfaces are deprecated in Deno and that these utilities are deprecated along with them, and the module's recent history is almost entirely removals (`BufReader`, `BufWriter`, `readDelim`, `MultiReader`). Adding public surface to a module we're winding down is a policy call, so I'd like a second maintainer opinion before we take this. Please don't take that as a rejection — it's a question about `@std/io` rather than about your change.

If we do take it, I've left inline notes on a few things that should be fixed first. The `NaN` one is the important one — it silently corrupts the buffer.

One more, on naming: `readDiscard` doesn't match anything else on the class (`truncate`, `reset`, `bytes`, `empty`). Since you invited suggestions — `discard(n)` or `skip(n)` reads better to me.
